jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,43 +119,160 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String imagesDir;
+	private boolean leaveRunning;
+	private boolean shellJob;
+	private boolean extUnixSupport;
+	private int logLevel;
+	private String logFile;
+	private boolean fileLocks;
+	private String workDir;
+
+	/**
+	 * Sets the directory that will hold the images upon dump. Corresponds to criu_set_images_dir_fd.
+	 * This must be set before calling {@link #checkPointJVM()}.
+	 * 
+	 * @param imagesDir the directory as a java.nio.file.Path
+	 * @return this
+	 * @throws NullPointerException if imagesDir is null.
+	 * @throws SecurityException if no permission to access imagesDir.
+	 */
+	public CRIUSupport setImagesDir(Path imagesDir) {
```

Comment:
can you do a test to see if you can actually create a checkpoint image without setting the imageDir. If it is not possible then the constructor should take `imagesDir` as a param, it should not be optional. AFAIK all the other params are optional.

Reply:
also we need to validate that the directories are actaully directories, look at `Files::isDirectory`

Reply:
I hope that doesn't isn't so restrictive that it disallows use of a symbolic link (that refers to a directory).

Reply:
When I comment all the imageDir stuff out from criusupport.c: 
```
Error (criu/cr-service.c:379): Can't chdir to work_dir: Not a directory                                            
res.getType() SYSTEM_CHECKPOINT_FAILURE 
```
Added param and docs to constructor

Reply:
> I hope that doesn't isn't so restrictive that it disallows use of a symbolic link (that refers to a directory).

```
<p> The {@code options} array may be used to indicate how symbolic links
are handled for the case that the file is a symbolic link. By default,
symbolic links are followed and the file attribute of the final target
of the link is read. If the option {@link LinkOption#NOFOLLOW_LINKS
NOFOLLOW_LINKS} is present then symbolic links are not followed.
```

Reply:
Added checks for valid dir and correct exceptions are thrown when a file or a nonexistent dir is supplied to the constructor